django/contrib/gis/db/models/proxy.py

```diff
@@ -52,7 +52,9 @@ def __set__(self, obj, value):
 
         # The geometry type must match that of the field -- unless the
         # general GeometryField is used.
-        if isinstance(value, self._klass) and (str(value.geom_type).upper() == gtype or gtype == 'GEOMETRY'):
+        if gtype == 'RASTER':
```

Comment:
Also here, the class is called `GeometryProxy` but can be used for raster instantiation as well. Renaming would require deeper change in the codebase. Not sure what the better solution is, renaming or using it as is.